### PR TITLE
PLT-525 - Delegated Access - review fix

### PR DESCRIPTION
**Pull request by abdulrkalam**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppspersononprobationuserapi/service/DelegatedAccessService.kt [hunk 1]
```diff
@@ -20,30 +20,25 @@ class DelegatedAccessService(private val delegatedAccessRepository: DelegatedAcc
...
-        createdDate = now,
-        deletedDate = null,
-      )
-      return delegatedAccessRepository.save(delegatedAccessEntity)
-    } else {
-      throw ValidationException(
-        "Request invalid. " +
-          "initiatedUserId= ${delegatePost.initiatedUserId} " +
-          "delegatedUserId= ${delegatePost.delegatedUserId} ",
+    val initiatedUser = userRepository.findById(delegatePost.initiatedUserId.toLong()) ?: throw ResourceNotFoundException("User with id ${delegatePost.initiatedUserId} not found in database ")
```
Comment (mkerry-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppspersononprobationuserapi/service/DelegatedAccessService.kt:23]: adding the null check on the findById that returns Optional<UserEntity> won't work. You can call `getOrNull()` on it first

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppspersononprobationuserapi/service/DelegatedAccessService.kt [hunk 2]
```diff
@@ -20,30 +20,25 @@ class DelegatedAccessService(private val delegatedAccessRepository: DelegatedAcc
...
+    val delegatedUser = userRepository.findById(delegatePost.delegatedUserId.toLong()) ?: throw ResourceNotFoundException("User with id ${delegatePost.delegatedUserId} not found in database ")
 
+    val delegatedAccessAlreadyExists =
+      delegatedAccessRepository.findByInitiatedUserIdAndDelegatedUserIdAndDeletedDateIsNull(
+        initiatedUser.get().id!!.toLong(),
+        delegatedUser.get().id!!.toLong(),
```
Comment (mkerry-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppspersononprobationuserapi/service/DelegatedAccessService.kt:29]: we should declare the input as a long, else we may end up with unsubmittable user ids

Comment (abdulrkalam) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppspersononprobationuserapi/service/DelegatedAccessService.kt]: Done

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppspersononprobationuserapi/service/DelegatedAccessService.kt (hunk 1 above)
Comment (abdulrkalam) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppspersononprobationuserapi/service/DelegatedAccessService.kt]: Done

**Review by mkerry-moj — APPROVED** [review]

**Pull request merged**